src/blocks/weather.rs

```diff
@@ -190,21 +184,21 @@ pub async fn run(config: toml::Value, mut api: CommonApi) -> Result<()> {
         .new_widget()
         .with_format(config.format.with_default("$weather $temp")?);
 
-    // met_no requires a one-time fetch of a set of "legends", which are
-    // translations of symbols into weather strings.
-    let legends_store: met_no::LegendsStore = match config.service {
-        WeatherService::MetNo(_) => met_no::get_legend().await?,
-        _ => HashMap::new(),
+    let provider: Arc<Mutex<dyn WeatherProvider + Send + Sync>> = match config.service {
+        WeatherService::MetNo(config) => Arc::new(Mutex::new(config)),
+        WeatherService::OpenWeatherMap(config) => Arc::new(Mutex::new(config)),
     };
 
     loop {
-        let location = match config.autolocate {
-            true => find_ip_location().await.unwrap_or(None),
-            false => None,
-        };
-
         let data = api
-            .recoverable(|| config.service.get(&legends_store, &location))
+            .recoverable(|| async {
+                let location = match config.autolocate {
+                    true => find_ip_location().await.unwrap_or(None),
```

Comment:
Now that it is inside `api.recoverable`, this can/should propagate the error: `true => Some(find_ip_location().await?)`